src/hooks/useGasPrice.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react'
+import useSafeState from './useSafeState'
+import { walletApi } from 'api'
+
+export const useGasPrice = (defaultGasPrice: number | null = null): number | null => {
+  const [gasPrice, setGasPrice] = useSafeState<number | null>(null)
+  useEffect(() => {
+    async function getGasPrice(): Promise<void> {
+      const gasPrice = (await walletApi.getGasPrice()) || defaultGasPrice
```

Comment:
use API better

Reply:
I would prefer `walletApi.getGasPrice()` to be able to return `null` in case of failure, to better inform us whether it failed or not
And then we can decide to use default or not, in place where it really matters, like here